test/optimized-large-storage-test.js

```diff
@@ -0,0 +1,199 @@
+
+let writeTest = async function(fd,key,filesize,context){
+
+        const data = [];
+        for (let i = 0; i < filesize; i++) {
+             data.push(context);
+        }
+
+        let tx1 = await fd.write(key,data)
+        await tx1.wait()
+
+        let [resData,] = await fd.read(key)
+        // console.log(resData)
+
+        let [fsize,] = await fd.size(key)
+        // console.log(fsize)
+
+        expect(BigNumber.from(data).toHexString()).to.eq(resData)
+        expect(fsize.toNumber()).to.eq(filesize)
+}
+
+let writeChunkTest = async function(fd,key,chunkId,filesize,context) {
+
+        const data = [];
+        for (let i = 0; i < filesize; i++) {
+             data.push(context);
+        }
+
+        let tx1 = await fd.writeChunk(key,chunkId,data)
+        await tx1.wait()
+
+        let [resData,] = await fd.readChunk(key,chunkId)
+        // console.log(resData)
+
+        let [fsize,] = await fd.chunkSize(key,chunkId)
+        // console.log(fsize)
+
+        expect(BigNumber.from(data).toHexString()).to.eq(resData)
+        expect(fsize.toNumber()).to.eq(filesize)
+}
+
+let removeTest = async function(fd,key){
+    let tx1 = await fd.remove(key)
+    await tx1.wait()
+}
+
+let removeChunkTest = async function(fd,key,chunkId){
+    let tx1 = await fd.removeChunk(key,chunkId)
+    await tx1.wait()
+}
+
+let readTest = async function(fd,key,filesize1,context1,filesize2,context2,filesize3,context3) {
+
+    const data = [];
+    for (let i = 0; i < filesize1; i++) {
+         data.push(context1);
+    }
+    for (let i = 0; i < filesize2; i++) {
+        data.push(context2);
+    }
+    for (let i = 0; i < filesize3; i++) {
+        data.push(context3);
+    }
+    
+    let [resData,] = await fd.read(key)
+    let [fsize,] = await fd.size(key)
+
+    // console.log("Size:",fsize)
+    // console.log("Res Data:",resData)
+
+    if (data.length == 0){
+        // deal with special case: return data is null
+        expect(resData).to.eq("0x")
+        expect(fsize.toNumber()).to.eq(0)
+    }else{
+        expect(BigNumber.from(data).toHexString()).to.eq(resData)
+        expect(fsize.toNumber()).to.eq(filesize1+filesize2+filesize3)
+    }
+    
+}
+
+describe("OptimizedFlatDirectory Test", function () {
+
+    let OPFlatDirectory
+    beforeEach(async()=>{
+        factory = await ethers.getContractFactory("FlatDirectory");
+        OPFlatDirectory = await factory.deploy(220);
+        await OPFlatDirectory.deployed();
+    })
+
+    it("read/write",async function(){
+        await writeTest(OPFlatDirectory,"0x01",100,1)
+        await writeTest(OPFlatDirectory,"0x02",1000,1)
+        await writeTest(OPFlatDirectory,"0x03",10000,1)
+    })
+
+    it("rewrite",async function(){
+        await writeTest(OPFlatDirectory,"0x01",100,1)
+        await writeTest(OPFlatDirectory,"0x01",1000,2)
+        await writeTest(OPFlatDirectory,"0x01",10000,3)
+    })
+
+    it("writeChunk",async function(){
```

Comment:
I feel we could add a writeVariableSizedChunk test, which tests the size of the data can be from 0 to 300 (i.e, 301 tests) so that all cases are covered.

Reply:
like this  https://github.com/QuarkChain/evm-large-storage/pull/1/commits/ca0fc8553f69fbe91cd25828f986654cf855f54b  ?